plugins/main/macvlan/macvlan.go

```diff
@@ -51,13 +51,33 @@ func init() {
 	runtime.LockOSThread()
 }
 
+func getDefaultRouteInterfaceName() string {
+	routeToDstIP, err := netlink.RouteList(nil, netlink.FAMILY_ALL)
+	if err != nil {
+		return ""
+	}
+
+	for _, v := range routeToDstIP {
+		l, _ := netlink.LinkByIndex(v.LinkIndex)
+		if v.Dst == nil {
+			return l.Attrs().Name
+		}
+	}
+
+	return ""
```

Comment:
Would be nice to pass the errors up: the caller gives a generic one: would be nice to know whether there was no default route, or whether netlink gave an error etc.

Reply:
thank you for your comments. I changed it.